### PR TITLE
added function for phone number parser and test

### PR DESCRIPTION
**Pull request by bookwyrm31415**
## Proposed Changes
1.  Added a phone number parses in the utils folder, and a few tests for it

## Additional Info.
The parser function either returns a string of the phone number you can put in the href link or null if the string does not have a phone number in it

### CODE REVIEW

**Comment by noway** [comment]
OMG, awesome!! I'll have a look tonight

**Comment by bookwyrm31415** [comment]
It's just a wrapper for a bunch of regex to handle all the edge cases on
the issue page.

I hope thats kind of what you wanted 🤷‍♂️

If not let me know and I can change it

File: src/utils/parsePhone.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+
+//Takes in a string and returns either the forst valid phone number or null
+
```
Comment (olafwrieden) [inline src/utils/parsePhone.ts:4]: Please reduce the whitespace here, you can condense this to one line.

File: src/utils/parsePhone.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { parsePhoneNumber } from "./parsePhone";
+
+describe("Normal number", () => {
+    it("should return a string of the normal phone number", () => {
+
+        const want = "(07) 873 7079";
+
+        const got = parsePhoneNumber("{ telephone: '(07) 873 7079  /  A/H (07) 873 8399' }");
+
```
Comment (olafwrieden) [inline src/utils/parsePhone.spec.ts:9]: Reduce whitespace please, no need for blank lines in between. Tip: Also run prettier for formatting. We use an indentation of tab size: 2 throughout this project (not 4 spaces indentation) - Prettier should fix this for you.

File: src/utils/parsePhone.spec.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    it("should return a null", () => {
+
+        const want = null;
+
+        const got = parsePhoneNumber("test string no actual phone number");
+
+        expect(got).toBe(want);
+    })
+
+
```
Comment (olafwrieden) [inline src/utils/parsePhone.spec.ts:34]: Reduce whitespace please. Only add 1 blank line between test cases,

File: src/utils/parsePhone.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+export function parsePhoneNumber(potentialNumber: string): string | null {
+    let phoneRegex = /[+]*[(]{0,1}[0-9]{1,4}[)]{0,1}[-\s\.0-9]*/
+    let eightHundredRegex = /0800\s[A-Z]+/
+    //Does a standard phone regex, but that will also just return the 0800 from a number like 0800 ORANGA
+    //So I check its length and do an 0800 regex as well
```
Comment (olafwrieden) [inline src/utils/parsePhone.ts:9]: Small ask to improve the comment, remember other people need to read this in the context of what the code does, not what you ("so _I_ check") do. Thanks!

**Review by olafwrieden — CHANGES_REQUESTED** [review]
@bookwyrm31415 I'd like to request small formatting changes. From an implementation point of view, are you able to implement your function on all the modals where phone numbers are displayed? We'd need that to ship this functionality and close the original issue. Love this!

**Comment by bookwyrm31415** [comment]
So sorry about all the intermediate commits, but this latest one passes the
checks

Regards
Ryan

**Review by olafwrieden — APPROVED** [review]
Looks good to me. @noway can you give it glance too and merge when convenient. Thanks! Closes #167

**Review by noway — APPROVED** [review]
Love it ❤️ . I've added 1 more test from the original issue. All works nicely. Good work  @bookwyrm31415

**Pull request merged**